aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -36,12 +36,12 @@ static inline Tensor integer_upcast(const Tensor& self, optional<ScalarType> dty
 
 using DimMask = TensorIterator::DimMask;
 
-static DimMask make_dim_mask(IntList dims, int ndim) {
+static DimMask make_dim_mask(IntList dims, int64_t ndim) {
   auto mask = DimMask();
   if (dims.empty()) {
     mask.flip();
   } else {
-    for (int dim : dims) {
+    for (size_t dim : dims) {
```

Comment:
this doesn't look right, shouldn't it be `int64_t` because that's the element type of `IntList`?

Reply:
Good catch, the fact that the vector's length is size_t while it holds int64_t is tripping me up :). Will update this and the others